subsys/bootloader/image/CMakeLists.txt

```diff
@@ -4,59 +4,51 @@
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 #
 
+if (CONFIG_BOOTLOADER_MCUBOOT)
+  set(image_to_boot mcuboot_)
+  set(s0_addr $<TARGET_PROPERTY:partition_manager,PM_S0_ADDRESS>)
+  set(s1_addr $<TARGET_PROPERTY:partition_manager,PM_S1_ADDRESS>)
+else()
+  if (CONFIG_SPM)
+    set(image_to_boot spm_)
+    set(s0_addr $<TARGET_PROPERTY:partition_manager,PM_SPM_ADDRESS>)
+  else()
+    set(app_CONFIG_BOOTED_BY_SECURE_BOOT "y" CACHE STRING "Booted by B0")
```

Comment:
Why is this not needed for the other cases?

Reply:
This should not be here. This is now solved by updating `${image_to_boot}OVERLAY_CONFIG`